### ISSUE TITLE
Unable to open a version

### ISSUE THREAD

**Issue by mlstoppa** [ARCAD-Software/elias-vscode#41]
Hi @sebjulliand 
It looks like with the latest version of Elias software it's not possible to open a version anymore.
I assume it's because an older Arcad version is installed on the IBMi (Version 13.02.11), so some parameters are not allowed yet.
This is the log:

```
[8/29/2024 11:45:25 AM] [TRC] REST request ARCAD-Elias REST (http) lstoppa@developm.cust.net: post /command/async/AOPNVER  {"parameters":{"APP":"'EUR'","ENV":"'DEV'","VERSION":"*GEN","VERTYPE":"*RELEASE","INCREMENT":"*VERTYPE","TEXT":"'JDA-2452 Fix SRV001.ParmsSendMail bug'","CHKFROMREF":"*VERTYPE","LOCKOBJ":"*VERTYPE","ALWEDITSRC":"*VERTYPE","XREFLVL":"*VERTYPE","UPDXREF":"*VERTYPE","VERCHG":"*VERTYPE","VERSTS":"OPEN","CRTBRANCH":"*NO","SCMREBASE":"*NO"}}
[8/29/2024 11:45:27 AM] [TRC] REST response ARCAD-Elias REST (http) /command/async/AOPNVER lstoppa@developm.cust.net 200 - OK: {"data":{"command":"SBMJOB CMD(*LIBL/AOPNVER APP('EUR') TEXT('JDA-2452 Fix SRV001.ParmsSendMail bug') ENV('DEV') CHKFROMREF(*VERTYPE) VERCHG(*VERTYPE) ALWEDITSRC(*VERTYPE) INCREMENT(*VERTYPE) CRTBRANCH(*NO) SCMREBASE(*NO) LOCKOBJ(*VERTYPE) VERSION(*GEN) XREFLVL(*VERTYPE) UPDXREF(*VERTYPE) VERTYPE(*RELEASE) VERSTS(OPEN)) JOBQ(*JOBD) USER(*CURRENT) JOB(AOPNVER) HOLD(*YES)","messages":[{"from_instruction":"07D7","from_library":"","from_program":"QCAPOS","messageFile":"QCPFMSG","message_text":**"Keyword CRTBRANCH not valid for this command."**,"message_time":"2024-08-29-02.45.27.0000","msgid":"CPD0043","second_level":"Cause . . . . . :   A keyword was specified on a command that has no corresponding parameter. Recovery  . . . :   Correct or omit the keyword, and then try the command again.","severity":30,"to_instruction":"0000","to_module":"","to_procedure":"","to_program":"QZRCSRVS","type":"DIAGNOSTIC"},{"from_instruction":"07D7","from_library":"","from_program":"QCAPOS","messageFile":"QCPFMSG","message_text":**"Keyword SCMREBASE not valid for this command."**,"message_time":"2024-08-29-02.45.27.0000","msgid":"CPD0043","second_level":"Cause . . . . . :   A keyword was specified on a command that has no corresponding parameter. Recovery  . . . :   Correct or omit the keyword, and then try the command again.","severity":30,"to_instruction":"0000","to_module":"","to_procedure":"","to_program":"QZRCSRVS","type":"DIAGNOSTIC"},{"from_instruction":"0142","from_library":"","from_program":"QCAFCMD","messageFile":"QCPFMSG","message_text":"Previous 2 errors found in embedded command AOPNVER.","message_time":"2024-08-29-02.45.27.0000","msgid":"CPD0099","second_level":"Cause . . . . . :   The previous 2 errors were detected in an embedded command. Recovery  . . . :   Correct the errors in the embedded command.","severity":30,"to_instruction":"0000","to_module":"","to_procedure":"","to_program":"QZRCSRVS","type":"DIAGNOSTIC"},{"from_instruction":"0544","from_library":"","from_program":"QCADRV2","messageFile":"QCPFMSG","message_text":"Error found on SBMJOB command.","message_time":"2024-08-29-02.45.27.0000","msgid":"CPF0001","second_level":"Cause . . . . . :   The system detected errors in the command. Recovery  . . . :   See the previously listed messages in the job log. Correct the errors and then try the command again.  *N instead of a command name means that the name had not been determined before the error was found.","severity":30,"to_instruction":"0000","to_module":"","to_procedure":"","to_program":"QZRCSRVS","type":"ESCAPE"}],"successful":false}}
[8/29/2024 11:45:27 AM] [ERR] Failed to open version
SBMJOB CMD(*LIBL/AOPNVER APP('EUR') TEXT('JDA-2452 Fix SRV001.ParmsSendMail bug') ENV('DEV') CHKFROMREF(*VERTYPE) VERCHG(*VERTYPE) ALWEDITSRC(*VERTYPE) INCREMENT(*VERTYPE) CRTBRANCH(*NO) SCMREBASE(*NO) LOCKOBJ(*VERTYPE) VERSION(*GEN) XREFLVL(*VERTYPE) UPDXREF(*VERTYPE) VERTYPE(*RELEASE) VERSTS(OPEN)) JOBQ(*JOBD) USER(*CURRENT) JOB(AOPNVER) HOLD(*YES)
[CPD0043] [30] Keyword CRTBRANCH not valid for this command.
	Cause . . . . . :   A keyword was specified on a command that has no corresponding parameter. Recovery  . . . :   Correct or omit the keyword, and then try the command again.
[CPD0043] [30] Keyword SCMREBASE not valid for this command.
	Cause . . . . . :   A keyword was specified on a command that has no corresponding parameter. Recovery  . . . :   Correct or omit the keyword, and then try the command again.
[CPD0099] [30] Previous 2 errors found in embedded command AOPNVER.
	Cause . . . . . :   The previous 2 errors were detected in an embedded command. Recovery  . . . :   Correct the errors in the embedded command.
[CPF0001] [30] Error found on SBMJOB command.
	Cause . . . . . :   The system detected errors in the command. Recovery  . . . :   See the previously listed messages in the job log. Correct the errors and then try the command again.  *N instead of a command name means that the name had not been determined before the error was found.
```

Do you think it is possible to somehow manage the coexistence of older and newer versions?

Thank you very much!
Lucia

**Comment by sebjulliand** [comment]
Hi @mlstoppa ,
we could manage this for some simple cases like the one you showed. But we won't go as far as fully supporting versions of ARCAD that are this old 😅 (v13.2 is 3 years old!)
But a least for commands, I'll try to see if the command processor can manage these cases (like catching `CPD0043` and re-run the command without the unknown parameters). If it works, it can be integrated in a future major release of the server.

**Comment by sebjulliand** [comment]
The method described above works and will be released in version `2.2.0` (should be released during the second half of September).

**Comment by sebjulliand** [comment]
Version 2.2.0 is out 🎉
